Source/WebCore/workers/service/server/SWServerWorker.cpp

```diff
@@ -201,7 +201,7 @@ void SWServerWorker::didFinishInstall(const std::optional<ServiceWorkerJobDataId
         return;
 
     ASSERT(m_server);
-    RELEASE_ASSERT_WITH_MESSAGE(state == ServiceWorkerState::Installing, "State is %hhu", state);
+    RELEASE_ASSERT_WITH_MESSAGE(state == ServiceWorkerState::Installing, "State is %hhu", static_cast<int>(state));
```

Comment:
Let's cast it to uin8_t since state is enum uint8_t.
RELEASE_LOG though is able to use %hhu, do you know why this is the case for you but not here?

Reply:
It was strange for me too that it works in case of RELEASE_LOG and fails in case RELEASE_ASSERT_WITH_MESSAGE.

The difference in expanding this two macros are:

```
#define RELEASE_LOG(channel, ...) SD_JOURNAL_SEND(channel, LOG_NOTICE, __FILE__, _STRINGIFY(__LINE__), __func__, __VA_ARGS__)

// Expands to
do {
    if (LogServiceWorker.state != WTFLogChannelState ::Off)
        sd_journal_send_with_location("CODE_FILE="
                                      "/home/pgorszkowski/WebKit/Source/WebCore/workers/service/server/SWServerWorker.cpp",
            "CODE_LINE="
            "204",
            __func__, "WEBKIT_SUBSYSTEM=%s", LogServiceWorker.subsystem, "WEBKIT_CHANNEL=%s", LogServiceWorker.name, "PRIORITY=%i", 5, "MESSAGE="
                                                                                                                                       "State is %hhu",
            state, nullptr);
} while (0)
```

```
#define RELEASE_ASSERT_WITH_MESSAGE(assertion, ...) ASSERT_WITH_MESSAGE(assertion, __VA_ARGS__)

// Expands to
do {
    if (!(state == ServiceWorkerState ::Installing)) {
        WTFReportAssertionFailureWithMessage("/home/pgorszkowski/WebKit/Source/WebCore/workers/service/server/SWServerWorker.cpp", 206, __PRETTY_FUNCTION__, "state == ServiceWorkerState::Installing", "State is %hhu", state);
        WTFCrash();
    }
} while (0)
```

I could not find the reason of that. 

I checked and in case LOG with %hhu it also fails in debug.